src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -90,8 +101,8 @@ function setupScriptProcessor(recordBuffer, passSampleToVisualizers) {
 
       // While recording, feed data to recording buffer at the proper time.
       if (isRecording) {
-        recordBuffer
-            .copyToChannel(currentSamples[channel], channel, recordingLength);
+        let frameNumber = recordingLength / BUFFER_SIZE;
+        recordBuffer[channel][frameNumber] = currentSamples[channel].slice();
```

Comment:
What does this do?

Also `.slice()` creates a new instance. Is this intended?

Reply:
"=" in js make Object, array a shallow copy. We have to use slice to create a deep copy. Therefore, the data in recordingBuffer will be modified based on currentSamples[channel] update. 

There is another way is copy the float number in array one by one since "=" in js make number a deep copy. Due to the time and readability consuming, I think using 'slice' is better.

Reply:
Understood, but since this is happening around Float32Array, I recommend using `.set()` method for faster operation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set

It might need a bit or writing, but could you give some thoughts?

Reply:
Based on the reference, we only need to assign a new Float32Array. 

recordBuffer[channel][frameNumber] = new Float32Array(currentSamples[channel]);

In this way, we don't need set() OR slice(). I think these two way speed are similar. Basically, we are creating a new float32array.